app/src/main/java/com/sopt/now/presentation/auth/signup/SignupViewModel.kt

```diff
@@ -2,60 +2,39 @@ package com.sopt.now.presentation.auth.signup
 
 import android.app.Application
 import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
-import com.google.gson.Gson
-import com.sopt.now.data.auth.SignUpData.RequestSignUpDto
-import com.sopt.now.data.auth.SignUpData.ResponseSignUpDto
+import androidx.lifecycle.viewModelScope
 import com.sopt.now.data.ServicePool
+import com.sopt.now.data.auth.SignUpData.RequestSignUpDto
 import com.sopt.now.data.auth.SignUpData.SignUpState
-import retrofit2.Call
-import retrofit2.Callback
-import retrofit2.Response
+import com.sopt.now.presentation.auth.login.AuthRepository
+import kotlinx.coroutines.launch
 
 class SignUpViewModel(application: Application) : AndroidViewModel(application) {
-    private val authService by lazy { ServicePool.authService }
-    val liveData = MutableLiveData<SignUpState>()
+    private val authRepository by lazy { AuthRepository(ServicePool.authService) }
+    private val _signUpState = MutableLiveData<SignUpState>()
+    val signUpState: LiveData<SignUpState> = _signUpState
 
-    // 회원가입 함수
     fun signUp(request: RequestSignUpDto) {
-        authService.signUp(request).enqueue(object : Callback<ResponseSignUpDto> {
-            override fun onResponse(
-                call: Call<ResponseSignUpDto>,
-                response: Response<ResponseSignUpDto>,
-            ) {
-                if (response.isSuccessful) {
-                    val data: ResponseSignUpDto? = response.body()
-                    val userId = response.headers()["location"]
-                    liveData.value = SignUpState(
+        viewModelScope.launch {
+            val result = authRepository.signUp(request)
+            result.onSuccess { (memberId, responseBody) ->
+                _signUpState.postValue(
```

Comment:
```suggestion
            authRepository.signUp(
                request
            ).onSuccess { (memberId, responseBody) ->
                _signUpState.postValue(
```
result가 다른 곳에서 안 쓰인다면 따로 변수화 안 해줘도 바로 쓸 수 있을 것 같아요!

Reply:
헉 수정했습니당!